### PR TITLE
Update SQL schemas

### PR DESCRIPTION
**Pull request by csilvestrini**
Deleted `entities` table, added columns to `storage_key`, and a few
other missing columns.

Now we'll be able to store collections of primitives and singletons.

### CODE REVIEW

**Comment by csilvestrini** [comment]
It's possible I went a tad overboard with extension functions...

File: java/arcs/android/storage/database/DatabaseImpl.kt [hunk 1]
```diff
@@ -385,18 +383,43 @@ class DatabaseImpl(
...
+    }
+
+    private fun Boolean.toLong() = when (this) {
+        false -> 0L
+        true -> 1L
+    }
+
+    private fun Cursor.getBoolean(i: Int) = getLong(i).toBoolean()
+
+    private fun SQLiteProgram.bindBoolean(i: Int, value: Boolean) = bindLong(i, value.toLong())
```
Comment (jasonwyatt) [inline java/arcs/android/storage/database/DatabaseImpl.kt]: This are pretty great, maybe they could live in the `arcs.common.SqlUtils` file?   (would keep `Long.toBoolean`/`Boolean.toLong` private, and only have `Cursor.getBoolean` and `SQLiteProgram.bindBoolean` be public there)

File: java/arcs/android/storage/database/DatabaseImpl.kt [hunk 2]
```diff
@@ -385,18 +383,43 @@ class DatabaseImpl(
...
+
+    /** The type of the data stored at a storage key. */
+    @VisibleForTesting
+    enum class DataType {
+        ENTITY,
+        SINGLETON,
+        COLLECTION
```
Comment (jasonwyatt) [inline java/arcs/android/storage/database/DatabaseImpl.kt]: IIRC, kotlin style suggests using `CapitalCamelCase` for enum values.

Comment (csilvestrini) [reply java/arcs/android/storage/database/DatabaseImpl.kt]: Done

File: java/arcs/android/storage/database/DatabaseImpl.kt (hunk 1 above)
Comment (csilvestrini) [reply java/arcs/android/storage/database/DatabaseImpl.kt]: Done

**Pull request merged**